src/util.c

```diff
@@ -49,6 +49,53 @@ char *path_append(const char *path, const char *filename)
     return str;
 }
 
+char *escape_char(const char *url, const char escape_me)
```

Comment:
We are only escaping a single space character, please rename this function into ``escape_space`` or something. and remove the ``escape_me`` variable.